state/src/impls/top_level.rs

```diff
@@ -410,6 +410,15 @@ impl TopLevelState {
                     error: None,
                 })
             }
+            Action::Custom(bytes) => {
+                let handlers = self.db.custom_handlers().to_vec();
```

Comment:
Why do you convert it do Vec?

Reply:
`custom_handlers` returns reference, so `handlers` borrows `self`.
`self` is used as mutable reference in later code, so lifetime conflicts there.
So I wanted to clone `handlers`, and I thought `to_vec` would be a good solution.